### PR TITLE
Added support for torch.concat operator.

### PR DESCRIPTION
**Pull request by Michiel-Olieslagers**
Added torch.ops.aten.concatenate.default in the list of supported operators in the generic annotator. This allows torch.concat operators to be lowered to aten. 

Also added a test to check the functionality of the newly added operator to ensure functionality of added operator is preserved in future.

### CODE REVIEW

File: backends/arm/quantizer/quantization_annotation/generic_annotator.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
     torch.ops.aten.tile.default,
     torch.ops.aten.flip.default,
     torch.ops.aten.cat.default,
+    torch.ops.aten.concatenate.default,
```
Comment (digantdesai) [inline backends/arm/quantizer/quantization_annotation/generic_annotator.py:56]: Do we really see this op after export?

Comment (Michiel-Olieslagers) [reply backends/arm/quantizer/quantization_annotation/generic_annotator.py:56]: This issue was noticed during internal testing after export and thus this PR was needed to fix this issue.

**Comment by freddan80** [comment]
@Michiel-Olieslagers pls re-run CI since the PR merged this morning re-enables some tests. Thx

**Comment by Michiel-Olieslagers** [comment]
pull / unittest / linux / linux-job (pull_request) job is failing due to a URL 404 error not related to this PR.

**Review by freddan80 — APPROVED** [review]

**Pull request merged**